### PR TITLE
[WIP] Verison 1 'web_timeline' backport to support odoo8 views

### PR DESCRIPTION
**Pull request by nzaterrence**

### CODE REVIEW

**Comment by hbrunn** [comment]
did you check https://github.com/OCA/web/pull/306 ? this was the original PR of the module, and then abandoned because the author moved to a higher version

**Comment by pedrobaeza** [comment]
And the target is not correct. I retarget to 8.0.

**Comment by pedrobaeza** [comment]
This is totally messed, as the branch is not correct. You should start from 8.0 branch, not 9.0. Please fix it.

**Comment by thomaspaulb** [comment]
@nzaterrence I fixed the commits, but can you put the base branch back to 8.0?

**Comment by thomaspaulb** [comment]
@nzaterrence Getting an error on FormViewDialog, does not exist in 8.0. Can you check [this commit](https://github.com/OCA/web/pull/908/commits/fa454eb9aa6b9894938ca5e744d3780e97a40a1a#diff-2f26223a1c994010d8f4d139c73b438dR380) in which Pedro ported to 9.0, and undo some of it so that it works on 8.0 again.

File: web_timeline/static/src/js/web_timeline.js [hunk 1]
```diff
@@ -430,9 +430,10 @@ odoo.define('web_timeline.TimelineView', function (require) {
             }
             var data = {};
             data[self.fields_view.arch.attrs.date_start] =
-                time.auto_date_to_str(start, self.fields[self.fields_view.arch.attrs.date_start].type);
+                str_to_datetime(new moment(start).format('YYYY-m-DD hh:mm:ss'), self.fields[self.fields_view.arch.attrs.date_start].type);
```
Comment (thomaspaulb) [inline web_timeline/static/src/js/web_timeline.js]: For this one and also the end date, the month has to be a capital `M`.

File: web_timeline/static/src/js/web_timeline.js [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+            if (item.group != -1) {
+                group = item.group;
+            }
+            var data = {};
+            data[self.fields_view.arch.attrs.date_start] =
+                 str_to_datetime(new moment(start).format('YYYY-MM-DD hh:mm:ss'),
+                     self.fields[self.fields_view.arch.attrs.date_start].type);
+            data[self.fields_view.arch.attrs.date_stop] =
+                 str_to_datetime(new moment(end).format('YYYY-MM-DD hh:mm:ss'),
+                     self.fields[self.fields_view.arch.attrs.date_stop].type);
```
Comment (thomaspaulb) [inline web_timeline/static/src/js/web_timeline.js]: @nzaterrence Maybe something is going wrong here with the dragging of an element to new start date? Requires some deep debugging though.

**Comment by thomaspaulb** [comment]
@nzaterrence Can you check why the tests fail? And add any remaining fixes from the other copy? And then remove WIP status.

**Comment by pedrobaeza** [comment]
Is this going to be finished?

**Comment by thomaspaulb** [comment]
@pedrobaeza We are already using this for some time in production, together with a dependent timeline module for MRP work orders, which also uses the drag and drop functionality. Have updated the code and made it green.

**Comment by pedrobaeza** [comment]
OK, then I think we can merge it as this is for an old version. I would say to squash a bit the commit history, but as this is not going to be used for newer versions, we can keep it. Thanks for the work.

**Pull request merged**